util/common/src/main/java/org/hibernate/search/util/common/impl/Contracts.java

```diff
@@ -60,4 +60,10 @@ public static void assertNoNullElement(Collection<?> collection, String collecti
 			throw log.collectionMustNotContainNullElement( collectionDescription );
 		}
 	}
+
+	public static void assertTrue(boolean condition, String message) {
+		if ( !condition ) {
+			throw log.conditionFailed( message );
+		}
+	}
```

Comment:
Is that how it's done in Hibernate Validator?
I don't really see the added value compared to inlining the `if`, but if you feel strongly about it, I won't stop you...

Reply:
Yeah, this is how things are done in HV. I've been using a similar approach in other projects. But it's ok, you can stop me 😆